resources/aws_ec2.go

```diff
@@ -656,170 +656,66 @@ func (obj *AwsEc2Res) snsWatch() error {
 // CheckApply method for AwsEc2 resource.
 func (obj *AwsEc2Res) CheckApply(apply bool) (checkOK bool, err error) {
 	log.Printf("%s: CheckApply(%t)", obj, apply)
+	defer func() {
+		log.Printf("%s: CheckApply Done", obj)
+	}()
 
-	diInput := ec2.DescribeInstancesInput{
-		Filters: []*ec2.Filter{
-			{
-				Name:   aws.String("tag:Name"),
-				Values: []*string{aws.String(obj.prependName())},
-			},
-			{
-				Name: aws.String("instance-state-name"),
-				Values: []*string{
-					aws.String("running"),
-					aws.String("pending"),
-					aws.String("stopped"),
-					aws.String("stopping"),
-				},
-			},
-		},
-	}
-	diOutput, err := obj.client.DescribeInstances(&diInput)
+	// find the instance we need to check.
+	instance, err := describeInstanceByName(obj.client, obj.prependName())
 	if err != nil {
-		return false, errwrap.Wrapf(err, "error describing instances")
+		return false, errwrap.Wrapf(err, "error describing instance")
 	}
 
-	if len(diOutput.Reservations) < 1 && obj.State == "terminated" {
-		return true, nil
+	// If instance is nil, that means it's in a transitional state (pending,
+	// stopping, shutting-down), because an action was performed externally.
+	// We need to wait until the instance converges into a running, stopped,
+	// or terminated state, at which point we will receive an event.
+	if instance == nil {
+		log.Printf("%s: Could not apply, waiting for state to converge.", obj)
+		return false, nil
 	}
-	if len(diOutput.Reservations) == 1 && *diOutput.Reservations[0].Instances[0].State.Name == obj.State {
+
+	// if the instance is in the correct state, we're done.
+	if aws.StringValue(instance.State.Name) == obj.State {
```

Comment:
FYI, `aws.StringValue()` does the inverse of `aws.String()`. It dereferences the pointer "safely" by returning `""` if the pointer is `nil`